sisl/geometry.py

```diff
@@ -3005,13 +3006,30 @@ def scale(self, scale) -> Geometry:
 
         Parameters
         ----------
-        scale : float
+        scale : float or array-like of floats with shape (3,)
            the scale factor for the new geometry (lattice vectors, coordinates
            and the atomic radii are scaled).
+           If a vector is provided, each lattice vector is scaled using a different
+           scale factor, and therefore atomic radii don't get scaled since it doesn't
+           make sense.
         """
-        xyz = self.xyz * scale
-        atoms = self.atoms.scale(scale)
-        sc = self.sc.scale(scale)
+        if isinstance(scale, Iterable):
+            # The scale is a vector
+
+            # Atoms are not rescaled
+            atoms = self.atoms
```

Comment:
I would say that atoms should still be scaled according to the maximum scaling factor. If anything the `what` should be able to bypass this (I agree such an argument would make sense here).